client/my-sites/patterns/components/page-view-tracker.tsx

```diff
@@ -57,6 +59,7 @@ export function PatternsPageViewTracker( {
 	useEffect( () => {
 		if ( isDevAccount !== undefined && patternsCount !== undefined ) {
 			recordTracksEvent( 'calypso_pattern_library_view', {
+				name: patternPermalinkName,
```

Comment:
```suggestion
				name: patternPermalinkName || undefined,
```

This prevents us from sending an empty value to the `name` property. i.e. `name` will only be included in the tracking call if the name exists.

Reply:
We don't need it, since I implemented `patternPermalinkName` as `undefiend` if it's not permalink